docs/adr/adr-010-incentivized-testnet-monitoring.md

```diff
@@ -0,0 +1,139 @@
+
+We would like to make it possible for node operators to monitor their own nodes by deploying their own OLTP collector instance.
+
+This document proposes a strategy for making data in the Celestia team managed OTEL Collector available for use in internal Grafana dashboards and a public leaderboard. Additionally it describes how a node operator can monitor their own node by deploying their own OLTP collector instance.
+
+![Incentivized Testnet Monitoring Diagram](./img/incentivized-testnet-monitoring-diagram.svg)
+
+// Comment on diagram <https://lucid.app/lucidchart/d957570f-9c06-4a82-9843-00d8232f734a/edit?invitationId=inv_247247d6-3a67-40b1-8513-1818caadd627>#
+
+### Where to export data to?
+
+Grafana can query data from [multiple data sources](https://grafana.com/docs/grafana/latest/datasources/#supported-data-sources). This document explores two of these data sources:
+
+1. [Prometheus](https://github.com/prometheus/prometheus) is an open-source time series database written in Go. Prometheus uses the [PromQL](https://prometheus.io/docs/prometheus/latest/querying/basics/) query language. We can deploy Prometheus ourselves or use a hosted Prometheus provider (ex. [Google](https://cloud.google.com/stackdriver/docs/managed-prometheus), [AWS](https://aws.amazon.com/prometheus/), [Grafana](https://grafana.com/go/hosted-prometheus-monitoring/), etc.). Prometheus is pull-based which means services that would like to expose Prometheus metrics must provide an HTTP endpoint (ex. `/metrics`) that a Prometheus instance can poll (see [instrumenting a Go application for Prometheus](https://prometheus.io/docs/guides/go-application/)). Prometheus is used by [Cosmos SDK telemetry](https://docs.cosmos.network/main/core/telemetry.html) and [Tendermint telemetry](https://docs.tendermint.com/v0.35/nodes/metrics.html) so one major benefit to using Prometheus is that metrics emitted by celestia-core, celestia-app, and celestia-node can share the same database.
+1. [InfluxDB](https://github.com/influxdata/influxdb) is another open-source time series database written in Go. It is free to deploy the InfluxDB but there is a commercial offering from [influxdata](https://www.influxdata.com/get-influxdb/) that provides clustering and on-prem deployments. InfluxDB uses the [InfluxQL](https://docs.influxdata.com/influxdb/v1.8/query_language/) query language which appears less capable at advanced queries [ref](https://www.robustperception.io/translating-between-monitoring-languages/). InfluxDB is push-based which means services can push metrics directly to an InfluxDB instance ([ref](https://logz.io/blog/prometheus-influxdb/#:~:text=InfluxDB%20is%20a%20push%2Dbased,and%20Prometheus%20fetches%20them%20periodically.)). See [Prometheus vs. InfluxDB](https://prometheus.io/docs/introduction/comparison/#prometheus-vs-influxdb) for a more detailed comparison.
+
+If alternative data sources should be evaluated, please share them with us.
```

Comment:
Might worth mentioning [the Uptrace](https://uptrace.dev/docs/#getting-started). The reason is that its the most simple tool to use that covers tracing and metrics. It is like the Otel Collector, meaning you can easily connect the node. Still, the Collector is only a proxy and requires some other backends, while Uptrace is all-in-one solution.

Reply:
Honestly, I would even suggest using Uptrace for local setups of our user due to it simplicity

Reply:
I'll explore setting up Uptrace today and update / add to this section. Thanks for the suggestion.

Reply:
I set up Uptrace for metrics. Based on my limited usage, Uptrace doesn't seem as feature rich as Grafana for configuring dashboards. I think they may be more useful in exploring traces but I can't see any traces in Uptrace yet, have you been able to explore traces with https://github.com/celestiaorg/celestia-node/pull/907 ?

I'll push a commit soon that adds steps on how to setup Uptrace with celestia-node.

Reply:
Added section on Uptrace in [457908c](https://github.com/celestiaorg/celestia-node/pull/922/commits/457908ca6a519b309e1d63457782fb928dcd44b9)